### ISSUE TITLE
add `#![feature(negative_impls)]` but negative_impls doesnt exist localy

### ISSUE THREAD

**Issue by JAicewizard** [rust-lang/docs.rs#694]
**Crate name:**
rustt
**Build failure link:**
https://docs.rs/crate/rustt/0.1.1/builds/235564

**Additional details:**
The logs say I should use `#![feature(negative_impls)]` but 1: I cant find any information about this issue, and most importantly 2: locally using `nightly 2020-03-18` I cant build myself let alone upload to crates.io when adding this crate attribute.

**Comment by jyn514** [comment]
There is info about negative impls in https://github.com/rust-lang/rust/issues/68318 (there was a link in the long error message).

I'm confused how your crate is building locally without this feature but still having `impl !Trait {}`. Are you using nightly? Does `cargo doc` work locally?

**Comment by JAicewizard** [comment]
yes I am using nightly and it is documenting properly (without `#![feature(negative_impls)]`).
Adding `#![feature(negative_impls)]` makes it not doc and throw the error:
```
error[E0635]: unknown feature `negative_impls`
  --> src/lib.rs:80:12
   |
80 | #![feature(negative_impls)]
   |            ^^^^^^^^^^^^^^

error: aborting due to previous error

For more information about this error, try `rustc --explain E0635`.
error: Could not document `rustt`.

Caused by:
  process didn't exit successfully: `rustdoc --edition=2018 --crate-type lib --crate-name rustt src/lib.rs -o /home/jaap/Projects/rustt/target/doc --error-format=json --json=diagnostic-rendered-ansi -L dependency=/home/jaap/Projects/rustt/target/debug/deps --extern integer_encoding=/home/jaap/Projects/rustt/target/debug/deps/libinteger_encoding-9714e9561b194cec.rmeta --extern serde=/home/jaap/Projects/rustt/target/debug/deps/libserde-d23ce8522e605b3c.rmeta` (exit code: 1)
```
nightly version is `rustc 1.44.0-nightly (f509b26a7 2020-03-18)` which is the latest according to rustup.

**Comment by jyn514** [comment]
@JAicewizard I tried building this locally and got the same errors as from the docs.rs build:
```
error[E0658]: negative trait bounds are not yet fully implemented; use marker types for now
  --> src/ttv3/serde/mod.rs:40:6
   |
40 | impl !NotImplementKey for CustomNode {}
   |      ^^^^^^^^^^^^^^^^
   |
   = note: see issue #68318 <https://github.com/rust-lang/rust/issues/68318> for more information
   = help: add `#![feature(negative_impls)]` to the crate attributes to enable
```
Are you sure you're using 1.44 to document? I got this error with `rustdoc 1.47.0-nightly (6e87bacd3 2020-07-31)`, maybe try updating?

**Comment by JAicewizard** [comment]
I am very sure that at the time this was in deed the case!!!
currently on 46 nightly and I got the error now, and adding `#![feature(negative_impls)]` does actually work now.
I guess it was some kind of version difference, I don't know how since I tried updating a couple times (at the time).

**Comment by jyn514** [comment]
Very strange. Glad you figured it out though :) I see that you published 0.1.2 and it built so I'm going to close this.
